pkg/etcdcli/etcdcli.go

```diff
@@ -222,10 +228,38 @@ func (g *etcdClientGetter) MemberAdd(ctx context.Context, peerURL string) error
 		}
 	}
 
-	_, err = cli.MemberAdd(ctx, []string{peerURL})
+	defer func() {
+		if err != nil {
+			g.eventRecorder.Warningf("MemberAddAsLearner", "failed to add new member %v", peerURL)
```

Comment:
Yes, good catch. This would be an infrequent/notable event unlike member promotion failing for a short while.
I'll add the error here and in the other events.